cypher-shell/src/dist/cypher-shell

```diff
@@ -10,7 +10,8 @@ check_java() {
   [[ -n "${JAVA_MEMORY_OPTS:-}" ]] && version_command+=("${JAVA_MEMORY_OPTS[@]}")
 
   JAVA_VERSION=$("${version_command[@]}" 2>&1 | awk -F '"' '/version/ {print $2}')
-  if [[ "${JAVA_VERSION}" < "1.8" ]]; then
+  MAX_VERSION=`echo -e "$JAVA_VERSION\n1.8" | sort -t '.' -k 1,1 -k 2,2 -k 3,3 -k 4,4 -g | tail -1`
```

Comment:
I don't understand this line at all.

Reply:
It prints two lines, one with the current version and one with the minimum acceptable version, which it then sorts using each field in turn numerically (so 10 is greater than 1, etc.), and finally selects the last line after the sort. If the sorted last line in the minimum acceptable version, then the current version is smaller, and this is rejected in the following IF statement.